### ISSUE TITLE
Problem compiling CUDA code with new const

### ISSUE THREAD

**Issue by torsteingrindvik** [tracel-ai/cubecl#379]
In https://github.com/tracel-ai/cubecl/commit/f81a2fa775b5025a9794e4f159c7cd96765bc7da we got `const T * __restrict__` for readonly inputs which is great :tada: 

However it seems some patterns won't compile with this:

```
thread 'main' panicked at cubecl/crates/cubecl-cuda/src/compute/server.rs:380:17:
[Compilation Error]
    default_program(77): error: a value of type "const float *" cannot be used to initialize an entity of type "float *"
      float *slice_2_0 = input_2 + l_0_17;
```

So `input_2` is `const float* __restrict__ input_2`,

and it's being used like this:  `float *slice_2_0 = input_2 + l_0_17;`

I think the relevant kernel code is something like:

```rust
let slice = self.data.slice(start, start + foo);
```

So I think perhaps there is a missing `const` in the cpp source when trying to slice const inputs.

**Comment by nathanielsimard** [comment]
yeah we removed it because of that, I added a TODO in the code to add it back when we fix the slice notation to consider `const __restrict__`.
